pkg/wdk/certificate.go

```
@@ -19,10 +98,95 @@ type ListCertificatesResult struct {
 // extended with keyring and verifier
 type CertificateResult struct {
 	WalletCertificate
-	Keyring  map[primitives.StringUnder50Bytes]primitives.Base64String `json:"keyring"`
-	Verifier string                                                    `json:"verifier"`
+	Keyring  KeyringMap     `json:"keyring"`
+	Verifier VerifierString `json:"verifier"`
+}
+
+// VerifierString represents a string used as a verifier identifier or value.
+type VerifierString string
+
+// IsEmpty checks whether the VerifierString is empty.
+// Returns true if the string has length 0, otherwise false.
+func (s VerifierString) IsEmpty() bool {
+	return len(s) == 0
+}
+
+// parseSignature converts a HexString into an EC signature.
+// The input `s` is expected to be a concatenation of the R and S values in hex format (64 characters each).
+// Returns a pointer to an ec.Signature containing the parsed R and S values.
+// Otherwise error that indicates the given input string length is not correct.
+func parseSignature(s primitives.HexString) (*ec.Signature, error) {
+	if len(s) < 128 {
+		return nil, fmt.Errorf("input is too short to contain both R and S values (64 hex chars each)")
+	}
+
+	rHex := s[:64]
+	sHex := s[64:]
+
+	return &ec.Signature{
+		R: ec.FromHex(rHex.String()),
+		S: ec.FromHex(sHex.String()),
+	}, nil
+}
+
+// parseSerialNumber decodes a base64-encoded string into an sdk.SerialNumber.
+// Returns the decoded SerialNumber, or an error if decoding fails.
+func parseSerialNumber(s string) (sdk.SerialNumber, error) {
+	serialBytes, err := base64.StdEncoding.DecodeString(s)
+	if err != nil {
+		return sdk.SerialNumber{}, fmt.Errorf("failed to decode certificate serial number: %w", err)
+	}
+
+	var serial sdk.SerialNumber
+	copy(serial[:], serialBytes)
+
+	if len(serialBytes) != len(serial) {
+		return sdk.SerialNumber{}, fmt.Errorf("serial bytes length: %d is not equal to sdk.SerialNumber bytes length: %d", len(serialBytes), len(serial))
```

I would rather make this check "before" the copy phase.
The way I see it:
- The serial is a fixed size array (`32`) so if the length of `serialBytes` is not 32 - there is no point to proceed and we should return an error right away.